Feature branch

S-BLH opened the pull request:
1. Created a repository by the name of palindrome-checker.
2. cloned it to my local machine.
3. changed the branch from main to development and then feature-branch.
4. created index.html, style.css and script.js files.
5. configured the linters.
6. added the necessary code to complete the project.

S-BLH committed:
added html, css and js files

S-BLH committed:
added html,css and js linters

S-BLH committed:
added the html,css and js codes

S-BLH committed:
modified the errors

S-BLH committed:
modified the errors in the code

S-BLH committed:
modified the errors in the code

S-BLH committed:
added some changes

S-BLH committed:
changes to css were added

S-BLH committed:
added a pro readme file

S-BLH committed:
modified changes

S-BLH committed:
changes were added

S-BLH committed:
modified the changes

S-BLH committed:
modified some code errors

latifa-wakili commented:
great you created good repo
using git workflow
no linters error
clean codes 
keep it up!

ahzamir commented on index.html:
```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="UTF-8"></meta>
+  <meta name="viewport" content="width=device-width, initial-scale=1.0″></meta>
+  <title>Palindrome Checker</title>
+  <style>
+    body {
+      font-family: Arial, sans-serif;
+      text-align: center;
+    }
+  </style>
+</head>
```
- Your code has issues in here solve them:
![image](https://github.com/S-BLH/palindrome-checker/assets/96838030/1ee40f94-1d28-4ae5-bb2c-9cc2995df399)

ahzamir reviewed (changes requested):
@S-BLH 

- Well done you have implemented the most of project requirements but still your project is not complemented:

- Your project has no design and has issues in meta tags

S-BLH committed:
added background color

S-BLH commented:
> @S-BLH
> 
> * Well done you have implemented the most of project requirements but still your project is not complemented:
> * Your project has no design and has issues in meta tags

tnx sir now I implemented some designs and modified the errors, kindly review it back. @ahzamir

ahzamir commented on index.html:
```diff
@@ -0,0 +1,85 @@
+      margin: 0;
+      padding: 20px;
+      box-sizing: border-box;
+    }
+
+    h1 {
+      color: red;
+    }
+
+    label {
+      display: block;
+      font-size: 20px;
+      margin-bottom: 10px;
+    }
+
+    input[type="text"] {
+      width: 300px;
+      height: 30px;
+      font-size: 18px;
+      padding: 5px;
+    }
+
+    button {
+      background-color: red;
+      color: white;
+      font-size: 20px;
+      padding: 10px 20px;
+      border: none;
+      cursor: pointer;
+    }
+
+    button:hover {
+      background-color: darkred;
+    }
+
+    #result {
+      font-size: 24px;
+      margin-top: 20px;
+    }
+  </style>
```
- You have a separate file for your CSS code which is `style.css` move your styling code there.
  - Don't forget to link your CSS file in your HTML file using link tag in the head of your HTML file.

- Also your design is not responsive for mobile version please make your design responsive for the mobile version.

ahzamir commented on index.html:
```diff
@@ -0,0 +1,85 @@
+  <br><br>
+  <div id="result"></div>
+
+  <script>
+    const checkButton = document.getElementById('check-btn');
+    const resultDiv = document.getElementById('result');
+
+    checkButton.addEventListener('click', function() {
+      const userInput = document.getElementById('text-input').value;
+
+      if (userInput.trim().length === 0) {
+        alert('Please input a value');
+      } else {
+        const cleanedInput = userInput.toLowerCase().replace(/[^a-z0-9]/g, '');
+        const reversedInput = cleanedInput.split('').reverse().join('');
+        
+        if (cleanedInput === reversedInput) {
+          resultDiv.innerText = userInput + ' is a palindrome';
+        } else {
+          resultDiv.innerText = userInput + ' is not a palindrome';
+        }
+      }
+    });
+  </script>
```
- Same here you need to move your JS code to your `script.js` file and link your JS file using `<script>` tag at the end of your HTML body code.

ahzamir commented on README.md:
```diff
@@ -1,2 +1,24 @@
-# palindrome-checker
-first project of Java-script.
+# Palindrome-checker 🚀
+
+## Project Description 📝
+
+
+- 1sr project of javascript.
+- Its a palidrome cheker, it shows if a word is palidrome or not.
+- learnt new js,html and css codes and features.
+
+## Technologies Used 🛠️
+
+
+- HTML
+- CSS
+- JavaScript
+
+
+## Author 👩‍💻
+
+- LinkedIn: https://www.linkedin.com/in/sumita-balouch-959197309?utm_source=share&utm_campaign=share_via&utm_content=profile&utm_medium=ios_app
+- Email: sumitabalouch@gmail.com
+
+## License 📜
+MIT license.
```
- Add the Demo section with a screenshot and also add the Usage or How to use section and give users an instruction on how they can use your project.

ahzamir reviewed (changes requested):
@S-BLH  

- Well done you solved the code issues but still your project needs some changes t get the final approval please check the final comments for more details:

S-BLH committed:
separated the html,css and js from eachother and added a ss of the project

S-BLH committed:
modified the error in css

S-BLH committed:
solved the problem in js

S-BLH commented:
@ahzamir as you said sir, I made it responsive for all mobile versions and separated the html, css and js from e/o, kindly review it.

ahzamir reviewed (approved):
### Hi @S-BLH 🙋‍♀️,

> Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊

- You did not deployed your branch please deploy your code.
- **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.

- Share your Project on your LinkedIn account and add a proud moment text on it with a screenshot.

- Well Done great design Implemented. 🎯🎉👍
![image](https://github.com/S-BLH/palindrome-checker/assets/96838030/02fbbbfa-d896-40be-ad16-d121c4fc497d)

# Status: Approved ✅
There is nothing to say other than to merge your branch. 🎉🎯✅👍😊🥳

![WinkThumbsUpGIF](https://github.com/ZohalMirkhel/palindrome/assets/96838030/e5ee293e-072a-4806-bde0-b64e13140efa)

 ## Highlights 🌞

🎉 Great Design Implemented 
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)
✔️ You have followed the Palindrome Checker project task and committed using the command line
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔️Your code is clean
✔️Your repo has a professional README.md file
✔️Your repo have been reviewed by your partner
✔️Your design is responsive

**Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @ahzamir in your question so I can receive the notification.**
